lib/logstash/inputs/http_poller.rb

```diff
@@ -64,7 +67,57 @@ def register
 
     @logger.info("Registering http_poller Input", :type => @type,
                  :urls => @urls, :interval => @interval, :timeout => @timeout)
-  end # def register
+
+    setup_requests!
+  end
+
+  private
+  def setup_requests!
+    @requests = Hash[@urls.map {|name, url| [name, normalize_request(url)] }]
+  end
+
+  private
+  def normalize_request(url_or_spec)
+    if url_or_spec.is_a?(String)
+      res = [:get, url_or_spec]
+    elsif url_or_spec.is_a?(Hash)
+      # The client will expect keys / values
+      spec = Hash[url_or_spec.clone.map {|k,v| [k.to_sym, v] }] # symbolize keys
+
+      # method and url aren't really part of the options, so we pull them out
+      method = (spec.delete(:method) || :get).to_sym.downcase
+      url = spec.delete(:url)
+
+      # We need these strings to be keywords!
+      spec[:auth] = {user: spec[:auth]["user"], pass: spec[:auth]["pass"]} if spec[:auth]
```

Comment:
this field "pass" should be "password" since all other plugins seem to use "password" - 

```
% ack -h 'config :pass' ../logstash-*/lib/logstash/ | fex ,1 | sort | uniq -c
   1   config :passive
  25   config :password
```

Also, because the password here is a string, I wonder if normalize_request should wrap it up with a `LogStash::Util::Password` to protect it from being accidentally logged. Thoughts?